x/stakeibc/keeper/unbonding_records.go

```diff
@@ -61,23 +61,24 @@ func (k Keeper) SendHostZoneUnbondings(ctx sdk.Context, hostZone types.HostZone)
 			epochUnbondingRecordIds = append(epochUnbondingRecordIds, epochUnbonding.EpochNumber)
 			k.Logger(ctx).Info(fmt.Sprintf("[SendHostZoneUnbondings] Sending unbondings, host zone: %s, epochUnbonding: %v", hostZone.ChainId, epochUnbonding))
 
-
 		}
 	}
 	delegationAccount := hostZone.GetDelegationAccount()
 	if delegationAccount == nil || delegationAccount.GetAddress() == "" {
-		k.Logger(ctx).Error(fmt.Sprintf("Zone %s is missing a delegation address!", hostZone.ChainId))
-		return false
+		errMsg := fmt.Sprintf("Zone %s is missing a delegation address!", hostZone.ChainId)
+		k.Logger(ctx).Error(errMsg)
+		return nil, 0, nil, sdkerrors.Wrap(sdkerrors.ErrNotFound, errMsg)
 	}
 	validators := hostZone.GetValidators()
 	if totalAmtToUnbond == 0 {
-		return true
+		return nil, 0, nil, nil
```

Comment:
all of these return statements previously stopped the flow of the function, but now `SubmitHostZoneUnbondingMsg` _can_ execute, even if these conditions stop the flow (e.g. `totalAmtToUnbond == 0`). Thoughts on the best way to minimally change the logical flow? It might be as simple as adding some checks to `SubmitHostZoneUnbondingMsg`

Reply:
yeah good point -- adding the `nil` check on `msgs` in `SubmitHostZoneUnbondingMsg` should do it. It's only non-nil in the success case.